patches/chromium/replace_clearfilterdata_with_invalidatefilterdata.patch

```diff
@@ -0,0 +1,133 @@
+@@ -729,12 +721,8 @@ void SVGElementResourceClient::ResourceContentChanged(
+ 
+ void SVGElementResourceClient::ResourceElementChanged() {
+   if (LayoutObject* layout_object = element_->GetLayoutObject()) {
+-    ClearFilterData();
++    InvalidateFilterData();
+     SVGResourcesCache::ResourceReferenceChanged(*layout_object);
+-    // TODO(fs): If the resource element (for a filter) doesn't actually change
+-    // we don't need to perform the associated invalidations.
+-    layout_object->SetNeedsPaintPropertyUpdate();
+-    MarkFilterDataDirty();
+   }
+ }
+ 
+@@ -801,11 +789,16 @@ void SVGElementResourceClient::UpdateFilterData(
+   filter_data_dirty_ = false;
+ }
+ 
+-bool SVGElementResourceClient::ClearFilterData() {
+-  FilterData* filter_data = filter_data_.Release();
+-  if (filter_data)
++void SVGElementResourceClient::InvalidateFilterData() {
++  // If we performed an "optimized" invalidation via FilterPrimitiveChanged(),
++  // we could have set |filter_data_dirty_| but not cleared |filter_data_|.
++  if (filter_data_dirty_ && !filter_data_)
++    return;
++  if (FilterData* filter_data = filter_data_.Release())
+     filter_data->Dispose();
+-  return !!filter_data;
++  LayoutObject* layout_object = element_->GetLayoutObject();
++  layout_object->SetNeedsPaintPropertyUpdate();
++  MarkFilterDataDirty();
+ }
+ 
+ void SVGElementResourceClient::MarkFilterDataDirty() {
+diff --git a/third_party/blink/renderer/core/layout/svg/svg_resources.h b/third_party/blink/renderer/core/layout/svg/svg_resources.h
+index bfe056704b698d2189e5b759040f4f6cb3c54308..3416de3d5e062a31a0beb2df4d7398d2d435ec6a 100644
+--- a/third_party/blink/renderer/core/layout/svg/svg_resources.h
++++ b/third_party/blink/renderer/core/layout/svg/svg_resources.h
+@@ -228,7 +228,7 @@ class SVGElementResourceClient final
+                               const QualifiedName& attribute) override;
+ 
+   void UpdateFilterData(CompositorFilterOperations&);
+-  bool ClearFilterData();
++  void InvalidateFilterData();
+   void MarkFilterDataDirty();
+ 
+   void Trace(Visitor*) const override;
+diff --git a/third_party/blink/renderer/core/layout/svg/svg_resources_cache.cc b/third_party/blink/renderer/core/layout/svg/svg_resources_cache.cc
+index b9aea888de151de41591ec9fd76ef4a3c0f4b017..c94ad94190dcd280b80ed685955c94fedc5a077a 100644
+--- a/third_party/blink/renderer/core/layout/svg/svg_resources_cache.cc
++++ b/third_party/blink/renderer/core/layout/svg/svg_resources_cache.cc
+@@ -99,9 +99,8 @@ void SVGResourcesCache::ClientLayoutChanged(LayoutObject& object) {
+     invalidation_flags = SVGResourceClient::kBoundariesInvalidation;
+   bool filter_data_invalidated = false;
+   if (resources->Filter()) {
+-    filter_data_invalidated = client->ClearFilterData();
+-    invalidation_flags |=
+-        filter_data_invalidated ? SVGResourceClient::kPaintInvalidation : 0;
++    client->InvalidateFilterData();
++    invalidation_flags |= SVGResourceClient::kPaintInvalidation;
```

Comment:
Same question about `invalidation_flags`

Reply:
Just confirmed that `invalidation_flags` is indeed not needed here, and same with the previous comment above.